Formula/nlohmann-json.rb

```diff
@@ -26,30 +25,73 @@ def install
   end
 
   test do
-    (testpath/"test.cc").write <<~EOS
-      #include <iostream>
+    (testpath/"test.cpp").write <<~EOS
       #include <nlohmann/json.hpp>
 
       using nlohmann::json;
 
       int main() {
-        json j = {
+        // create an empty structure (null)
+        json j;
+
+        // add a number that is stored as double (note the implicit conversion of j to an object)
+        j["pi"] = 3.141;
+
+        // add a Boolean that is stored as bool
+        j["happy"] = true;
+
+        // add a string that is stored as std::string
+        j["name"] = "Niels";
+
+        // add another null object by passing nullptr
+        j["nothing"] = nullptr;
+
+        // add an object inside the object
+        j["answer"]["everything"] = 42;
+
+        // add an array that is stored as std::vector (using an initializer list)
+        j["list"] = { 1, 0, 2 };
+
+        // add another object (using an initializer list of pairs)
+        j["object"] = { {"currency", "USD"}, {"value", 42.99} };
+
+        // instead, you could also write (which looks very similar to the JSON above)
+        json j2 = {
           {"pi", 3.141},
+          {"happy", true},
           {"name", "Niels"},
+          {"nothing", nullptr},
+          {"answer", {
+            {"everything", 42}
+          }},
           {"list", {1, 0, 2}},
           {"object", {
-            {"happy", true},
-            {"nothing", nullptr}
+            {"currency", "USD"},
+            {"value", 42.99}
           }}
         };
-        std::cout << j << std::endl;
```

Comment:
Let's keep something like this in the test so we have output to check, please.

Reply:
Sure. I think I will bring back the old test. I realised that it's a better test for the Homebrew.